tests/integration/conftest.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2021 Canonical Ltd.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Fixtures for integration tests."""
+import os
+import pathlib
+import random
+import string
+import subprocess
+import sys
+import tempfile
+
+import pytest
+
+from craft_providers import lxd, multipass
+
+
+def generate_instance_name():
+    """Generate a random instance name."""
+    return "itest-" + "".join(random.choices(string.ascii_uppercase, k=8))
+
+
+@pytest.fixture()
+def home_tmp_path():
+    """Provide a temporary directory located in user's home directory.
+
+    Multipass doesn't have access to /tmp, this fixture provides an equivalent
+    to tmp_path for tests that require it.
+    """
+    with tempfile.TemporaryDirectory(
+        suffix=".tmp-pytest", dir=pathlib.Path.home()
+    ) as temp_dir:
+        yield pathlib.Path(temp_dir)
+
+
+@pytest.fixture()
+def instance_name():
+    """Provide a random name for an instance to launch."""
+    yield generate_instance_name()
+
+
+@pytest.fixture(scope="module")
+def reusable_instance_name():
+    """Provide a random name for an instance to launch with scope=module."""
+    yield generate_instance_name()
+
+
+@pytest.fixture(scope="module")
+def installed_lxd():
+    """Ensure lxd is installed, or skip the test if we cannot.
+
+    If the environment has CRAFT_PROVIDERS_TESTS_ENABLE_LXD_INSTALL=1,
+    force the installation of LXD if uninstalled.
+    """
+    if sys.platform != "linux":
+        pytest.skip(f"lxd not supported on {sys.platform}")
+
+    if lxd.is_installed():
+        return
+
+    if os.environ.get("CRAFT_PROVIDERS_TESTS_ENABLE_LXD_INSTALL") == "1":
+        lxd.install()
+    else:
+        pytest.skip("lxd not installed, skipped")
+
+
+@pytest.fixture
+def uninstalled_lxd():
+    """Uninstall Lxd prior to test, if environment allows it.
+
+    Environment may enable this fixture with:
+    CRAFT_PROVIDERS_TESTS_ENABLE_LXD_UNINSTALL=1
+    """
+    if sys.platform != "linux":
+        pytest.skip(f"lxd not supported on {sys.platform}")
+
+    if not lxd.is_installed():
+        pytest.skip("lxd not installed, skipped")
```

Comment:
if this fixture provides "an enviroment without LXD", shouldn't be a noop if LXD is just not installed, instead of making the test to skip?

Reply:
> if this fixture provides "an enviroment without LXD", shouldn't be a noop if LXD is just not installed, instead of making the test to skip?

It's for the same reason the lxd tests only run if lxd is installed.  If lxd is installed when tests are invoked, we assume lxd can run OK.

Reply:
Ok. Maybe worths a sentence in the function's docstring, something like "in case of LXD not being already installed, we assume that it cannot be installed and skip the test"

Reply:
Done!